api/app/Console/Commands/SyncPoolOperatorWithProcessOperator.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+namespace App\Console\Commands;
+
+use App\Models\Pool;
+use App\Models\Role;
+use App\Models\RoleAssignment;
+use App\Models\Team;
+use App\Models\User;
+use Illuminate\Console\Command;
+
+class SyncPoolOperatorWithProcessOperator extends Command
+{
+    /**
+     * The name and signature of the console command.
+     *
+     * @var string
+     */
+    protected $signature = 'app:sync-pool-process-operator';
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'Ensure all pool operator relations are represented in process operator relations';
+
+    /**
+     * Execute the console command.
+     *
+     * @return int
+     */
+    public function handle()
+    {
+        $poolOperatorRole = Role::where('name', 'pool_operator')->first();
+        $processOperatorRole = Role::where('name', 'process_operator')->first();
+
+        // collect all users that are pool operators
+        $poolOperatorUsers = User::whereHas('roleAssignments', function ($query) use ($poolOperatorRole) {
+            $query->where('role_id', $poolOperatorRole->id);
+        })->with('roleAssignments')->get();
+
+        foreach ($poolOperatorUsers as $poolUser) {
+            // collect array of teams the user is a pool operator on
+            $poolOperatorRoleAssignments = $poolUser->roleAssignments->where('role_id', $poolOperatorRole->id);
+            $poolOperatorTeams = ($poolOperatorRoleAssignments->pluck('team_id')->unique()->toArray());
+
+            // collect the pools per team then build an id array to sync
+            $poolIdsToSync = [];
+            foreach ($poolOperatorTeams as $teamId) {
+                $teamsPools = Pool::whereHas('legacyTeam', function ($query) use ($teamId) {
+                    $query->where('team_id', $teamId);
+                })->get();
+
+                $teamPoolsId = $teamsPools->pluck('id')->unique()->toArray();
+                array_push($poolIdsToSync, ...$teamPoolsId);
+            }
+
+            $uniquePoolIdsToSync = array_unique($poolIdsToSync);
+
+            // run through each pool id
+            foreach ($uniquePoolIdsToSync as $poolId) {
+                // create or fetch the pool teamable
+                $poolTeam = Team::firstOrCreate(
```

Comment:
I recommend using pool->addProcessOperators here, to help ensure consistency in how teams are created.

Reply:
Though... double check that it doesn't create duplicate roles for the same user-role-team combo, and fix it if it does 😅

Reply:
Ah, that would be a better idea. More condense too.
Seems to work fine with multiple runs, not spotting duplicates in table role_user
https://github.com/GCTC-NTGC/gc-digital-talent/pull/10743/commits/65a7f4992f0b860e3e27a39377ec0f913d89d49c